Products/PasswordStrength/backward.py

```diff
@@ -0,0 +1,64 @@
+from Products.PluggableAuthService.interfaces.plugins import IValidationPlugin
+from Products.CMFCore.utils import getToolByName
+from Products.PasswordStrength import _
+from plone.app.users.browser.register import BaseRegistrationForm
+from plone.app.users.browser.personalpreferences import PasswordAccountPanel
+
+
+# monkey patch to be used for Plone < 4.3
+def testPasswordValidity(self, password, confirm=None):
+
+    """ Verify that the password satisfies the portal's requirements.
+
+    o If the password is valid, return None.
+    o If not, return a string explaining why.
+    """
+    if confirm is not None and confirm != password:
+        return _('Your password and confirmation did not match. Please try again.')
+
+    if not password:
+        err = [_('You must enter a password')]
+    else:
```

Comment:
This is an unnecessary else. The contents of the following block are the initial state and the if statement and its block are modifying the initial state.